src/vcpkg/base/parse.cpp

```diff
@@ -105,6 +127,68 @@ namespace vcpkg
     {
     }
 
+    StringView ParserBase::skip_whitespace() { return match_while(is_whitespace); }
+    StringView ParserBase::skip_tabs_spaces()
+    {
+        return match_while([](char32_t ch) { return ch == ' ' || ch == '\t'; });
+    }
+
+    void ParserBase::skip_to_eof() { m_it = m_it.end(); }
+    void ParserBase::skip_newline()
+    {
+        if (cur() == '\r') next();
+        if (cur() == '\n') next();
+    }
+    void ParserBase::skip_line()
+    {
+        match_until(is_lineend);
+        skip_newline();
+    }
+
+    bool ParserBase::require_character(char ch)
+    {
+        if (static_cast<char32_t>(ch) == cur())
+        {
+            next();
+            return false;
+        }
+
+        add_error(msg::format(msgExpectedCharacterHere, msg::expected = ch));
+        return true;
+    }
+
+    bool ParserBase::try_match_keyword(StringView keyword_content)
+    {
+        auto first1 = m_it;
+        const auto last1 = first1.end();
+
+        auto first2 = keyword_content.begin();
+        const auto last2 = keyword_content.end();
+
+        for (;;)
+        {
+            if (first2 == last2)
+            {
+                if (first1 == last1 || is_whitespace(*first1))
+                {
+                    m_it = first1;
+                    m_column += static_cast<int>(keyword_content.size());
+                    return true;
+                }
+
+                return false;
+            }
+
+            if (first1 == last1 || *first1 != static_cast<char32_t>(*first2))
+            {
+                return false;
+            }
+
+            ++first1;
+            ++first2;
+        }
```

Comment:
```suggestion
        auto s_first = m_it.pointer_to_current();
        auto s_last = m_it.end().pointer_to_current();
        
        if (Strings::starts_with(StringView{s_first, s_last}, keyword_content))
        {
            auto end_of_keyword = s_first + keyword_content.size();
            if (end_of_keyword == s_last || is_whitespace(*end_of_keyword))
            {
                m_it = Utf8Decoder{end_of_keyword, s_last};
                m_column += static_cast<int>(keyword_content.size());
                return true;
            }
        }
        return false;
```

Reply:
I think this is technically a change in behavior because is_whitespace is operating in the encoding unit domain rather than the code point domain, so whitespace like `U+2029 PARAGRAPH SEPARATOR` can't be interpreted as whitespace.

I don't necessarily care about that but want to make sure you saw and considered this before applying.

Reply:
I did not apply this change because `m_it.end().pointer_to_current()` does not compile

Reply:
hmm, point. Can you do

```cxx
auto end_of_keyword = Utf8Decoder{s_first + keyword_content.size(), s_last};
if (end_of_keyword.is_eof() || is_whitespace(*end_of_keyword))
{
    m_it = end_of_keyword;
    ...
}
```

?

Reply:
That's getting into territory where I'm 'scared' about the invariants of Utf8Decoder; the original solution carefully avoids crossing from decoded world back into encoding unit world. Let me try adding some comments and see if you like that...

Reply:
Please let me know what you like of the solution I just pushed in https://github.com/microsoft/vcpkg-tool/pull/345/commits/19ae76dad6fecfb458764416a8346ddde836e6b6

Reply:
yeah, wfm